longoverdue.py

```diff
@@ -170,44 +177,54 @@ def restart():
     userservices = set()
     services = set()
     for proc in procs:
-        if proc.uunit != "-":
-            if proc.uunit.endswith(".service"):
-                userservices.add(proc.uunit)
-        elif proc.unit.endswith(".service"):
-            if proc.unit not in NO_AUTORESTART:
-                services.add(proc.unit)
+        if proc:
+            if proc.uunit != "-":
+                if proc.uunit.endswith(".service"):
+                    userservices.add(proc.uunit)
+            elif proc.unit.endswith(".service"):
+                if proc.unit not in NO_AUTORESTART:
+                    services.add(proc.unit)
 
     command = []
     if euid == 0 and services:
-        command = ["systemctl", "daemon-reload"]
-        print(" ".join(command))
-        subprocess.run(command, check=True)
-
-        command = ["systemctl", "restart"] + list(services)
-        print(" ".join(command))
-        subprocess.run(command, check=True)
+        try:
+            command = ["systemctl", "daemon-reload"]
+            print(" ".join(command))
+            subprocess.run(command, check=True)
+
+            command = ["systemctl", "restart"] + list(services)
+            print(" ".join(command))
+            subprocess.run(command, check=True)
+        except subprocess.CalledProcessError as e:
+            print("systemd command returned non-zero:", e)
```

Comment:
I'm not sure this is particularly valuable compared to just crashing with a stack trace. Have you ever run into a failure here? In that case, I think a more helpful error message would be better.

Reply:
Well, I'm open for a more detailed message. Certainly a one-line message should be preferable to a stack trace which 90% is useless information and takes up significantly more space.

Reply:
Comparing the new message and the stack trace, missing is only the command line. That certainly can be added as well, what do you think?

Reply:
We're already printing the command before running it so that won't be necessary. Considering these are the only two places where this exception isn't handled, and the message for getprocs also discards everything, I'll take it.
I suppose systemctl itself printed an error message of its own? If that's the case, I'd be okay with leaving it at that and not printing anything else, just `exit(1)` will be enough.

Reply:
Lets go with exit then.